producer.go

```diff
@@ -152,7 +151,51 @@ func (p *segmentProducer) configureSecurity(cfg Config) error {
 	return nil
 }
 
-func (p *segmentProducer) Send(key []byte, payload []byte, id []byte, ackFunc sdk.AckFunc) error {
+func (p *segmentProducer) Send(ctx context.Context, key []byte, payload []byte, id []byte, ackFunc sdk.AckFunc) error {
+	sendErr := p.sendRetryable(ctx, key, payload, id, ackFunc)
```

Comment:
Nitpick: it would be nicer to set the `ackFunc` in this function instead of passing it down to `sendRetryable` and doing it for every retried call (seems redundant). The nice benefit is also that only this function then cares about adding/removing `ackFunc` from the map.

Reply:
Good point about redundancy. When it comes to this function being the only one responsible for adding/removing ackFuncs from the map, then we still cannot achieve that, since we also need to remove the ackFuncs in the callback, when we actually get a response from Kafka about our write request.

Reply:
Well better to have it in 2 places than 3 then 😄